### PR TITLE
Add attributes to uptime module (CNAP 2083)

### PR DESCRIPTION
**Pull request by ReissJJarvis05**
Added new attributes to the create uptime check code and improved their descriptions to make their purposes more clear.

### CODE REVIEW

File: plugins/modules/uptime_check.py [hunk 1]
```diff
@@ -21,31 +21,59 @@
...
+            - The user can connect integrations which have been set up in the UI to the uptime check by specifying the integration IDs as a list of integers.
+    url:
+        required: false
+        description:
+            - A path on the destination server can be set for the uptime check to target. This is taken as a string.
     pause:
         required: false
         description:
-            - Not Required. Please set to "y" to pause the check on creation for testing
+            - To pause the uptime check on creation, set the value to "y". If left blank, the uptime check will begin running on creation.
```
Comment (andrew-garner) [inline plugins/modules/uptime_check.py]: I think "y" and blank are unusual values. Something like this should really be a bool and accept True or False, with a default of False. 

We should also add types onto fields, there is a specification for what can be defined in the DOCUMENTATION string at https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fields

**Comment by andrew-garner** [comment]
We should think about what we want to document in the README.md too. This module is also documented there and the parameters documented there are out of date. I'm reluctant to document things twice as it makes it easy to miss things and incorrect documentation is worse than not having any at all. 

I'm inclined to list the modules and lookups in the README, but instead of documenting all the variables there we add the appropriate command which will display the documentation from the module instead. What do you think?

File: plugins/modules/uptime_check.py [hunk 2]
```diff
@@ -21,31 +21,59 @@
     apikey:
         required: true
         description:
-            - api key to auth with Pingdom
-    url:
+            - The user's API key used to authorize the log in into Pingdom is taken as a string.
```
Comment (andrew-garner) [inline plugins/modules/uptime_check.py:25]: Please can you add the option type to each option, see https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fields

File: plugins/modules/uptime_check.py (hunk 1 above)
Comment (ReissJJarvis05) [reply plugins/modules/uptime_check.py]: Removed the code to accept "y" so pause now works as a boolean.

File: plugins/modules/uptime_check.py (hunk 2 above)
Comment (ReissJJarvis05) [reply plugins/modules/uptime_check.py:25]: Added "type" options to all attributes.

**Comment by andrew-garner** [comment]
Thanks for fixing the merge conflict. 

Please can you update the tests/test-uptime-check.yml file so it runs, and also convert the bool input for the pause parameter so there are no warnings printed when the test playbook runs. I get this when I run it...

```
TASK [Create uptime check] **************************************************************************************************************************
[WARNING]: The value "True" (type bool) was converted to "'True'" (type string). If this does not look like what you expect, quote the entire value
to ensure it does not change.
changed: [localhost]

TASK [debug] ****************************************************************************************************************************************
ok: [localhost] => 
  check:
    changed: true
    failed: false
    response:
      id: <redacted>
      name: OpenshiftAnsibleCheckTest created from localhost
    warnings:
    - The value "True" (type bool) was converted to "'True'" (type string). If this does not look like what you expect, quote the entire value to ensure it does not change.
```

File: plugins/modules/uptime_check.py [hunk 3]
```diff
@@ -61,11 +103,18 @@ def main():
         ## Set input variables
         fields = {
                 "apikey": {"type": "str", "required": True, "no_log": True},
-                "url": {"type": "str", "required": True},
+                "host": {"type": "str", "required": True},
```
Comment (andrew-garner) [inline plugins/modules/uptime_check.py:106]: The types here should match what is in the documentation above, which will mean converting all the non-string types into strings before calling pingdompy. Worth checking first what the api requires, as these might be easier to specify these as strings.

**Review by andrew-garner — CHANGES_REQUESTED** [review]

File: tests/test-uptime-check.yml [hunk 4]
```diff
@@ -1,45 +1,42 @@
...
-    check_tags: ocp-test,pypingdom-test
+    check_protocol: "http"
+    check_tags: "ocp-test,pypingdom-test"
     check_timing: "60"
+    check_port: "8080"
+    check_encryption: "true"
+    check_certificate: "true"
+    check_filters: "region:EU"
+    check_contain: "google"
+    check_ids: "88522"
```
Comment (andrew-garner) [inline tests/test-uptime-check.yml]: Please can you remove this ID, as it is specific to our Pingdom account.

**Review by andrew-garner — CHANGES_REQUESTED** [review]
One final change and I think we're there.

File: tests/test-uptime-check.yml (hunk 4 above)
Comment (ReissJJarvis05) [reply tests/test-uptime-check.yml]: Removed the ID

**Review by andrew-garner — APPROVED** [review]

**Pull request merged**